circus-train-core/src/test/java/com/hotels/bdp/circustrain/core/CopierFactoryManagerTest.java

```diff
@@ -60,6 +63,44 @@ public void supportsScheme() {
   public void doesNotSupportScheme() {
     when(copierFactory.supportsSchemes(SCHEME, SCHEME)).thenReturn(false);
 
-    copierFactoryManager.getCopierFactory(path, path);
+    copierFactoryManager.getCopierFactory(path, path, ImmutableMap.<String, Object>of());
+  }
+
+  @Test
+  public void supportsSchemeWithCopierFactoryClass() {
+    CopierFactory testCopierFactory = new TestCopierFactory();
+    copierFactoryManager = new CopierFactoryManager(Arrays.asList(testCopierFactory));
+    copierFactoryManager.postConstruct();
+
+    CopierFactory copierFactoryResult = copierFactoryManager.getCopierFactory(path, path, ImmutableMap.<String, Object>of("copier-factory-class", "com.hotels.bdp.circustrain.core.CopierFactoryManagerTest$TestCopierFactory"));
+
+    assertTrue(copierFactoryResult == testCopierFactory);
```

Comment:
AssertEquals would be better here, you get a better error message if the assertion fails showing what the values are rather than just "wasn't true".